### PR TITLE
Monix support

### PR DESCRIPTION
**Pull request by bfdes**
Attempt to address https://github.com/disneystreaming/weaver-test/issues/43.

### CODE REVIEW

File: modules/monix/src/weaver/monixcompat/suites.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import monix.eval.instances._
+import monix.execution.Scheduler
+
+trait BaseMonixSuite { self: ConcurrentEffectSuite[Task] =>
+  val ec                              = scala.concurrent.ExecutionContext.global
+  val scheduler                       = Scheduler(ec)
+  implicit def timer: Timer[Task]     = Task.timer(scheduler)
+  implicit def cs: ContextShift[Task] = Task.contextShift(scheduler)
+  implicit def effect: ConcurrentEffect[Task] =
+    new CatsConcurrentEffectForTask()(scheduler, Task.defaultOptions)
```
Comment (bfdes) [inline modules/monix/src/weaver/monixcompat/suites.scala:16]: It remains to see whether these defaults are sensible.

File: modules/monix/src/weaver/monixcompat/suites.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+trait MutableMonixSuite extends MutableFSuite[Task] with BaseMonixSuite
+
+trait SimpleMutableMonixSuite extends MutableMonixSuite {
+  type Res = Unit
+  def sharedResource: Resource[Task, Unit] = Resource.pure(())
```
Comment (bfdes) [inline modules/monix/src/weaver/monixcompat/suites.scala:23]: Still looking into whether `Bracket` should be used instead of `Resource`. From my understanding, Monix `Task.bracket` simply creates a Cats Effect `Bracket`...

File: modules/monix/test/src/weaver/monixcompat/MonixSuiteTest.scala [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import cats.effect.Resource
+
+import weaver.framework.DogFood
+
+import monix.eval.Task
+import sbt.testing.Status
+
+object MonixSuiteTest extends MonixSuite {
+  type Res = Unit
+  override def sharedResource: Resource[Task, Unit] = Resource.pure(())
```
Comment (bfdes) [inline modules/monix/test/src/weaver/monixcompat/MonixSuiteTest.scala:12]: TODO: Test resource access patterns as in ZIO combat tests.

File: modules/monix/src/weaver/monixcompat/suites.scala [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+package monixcompat
+
+import cats.effect.{ ConcurrentEffect, ContextShift, Resource, Timer }
+
+import monix.eval.Task
+import monix.eval.instances._
+import monix.execution.Scheduler
+
+trait BaseMonixSuite { self: ConcurrentEffectSuite[Task] =>
+  val ec                              = scala.concurrent.ExecutionContext.global
```
Comment (Baccata) [inline modules/monix/src/weaver/monixcompat/suites.scala:11]: I think I wouldn't expose this one publicly. `Scheduler` is more central to monix, and also extends ExecutionContext

File: modules/monix/src/weaver/monixcompat/suites.scala (hunk 1 above)
Comment (Baccata) [reply modules/monix/src/weaver/monixcompat/suites.scala:16]: I think it's okay

File: modules/monix/src/weaver/monixcompat/suites.scala (hunk 2 above)
Comment (Baccata) [reply modules/monix/src/weaver/monixcompat/suites.scala:23]: Let's leave it as Resource. Monix doesn't have (afaik) a competing construct to `cats.effect.Resource`(zio however does). `Resource` makes use of the `Bracket` typeclass, which is implemented using `Task#bracket` in this case

File: modules/monix/test/src/weaver/monixcompat/MonixSuiteTest.scala (hunk 3 above)
Comment (Baccata) [reply modules/monix/test/src/weaver/monixcompat/MonixSuiteTest.scala:12]: Not necessary 😄 , you're extending `MutableFSuite` which is already tested. zio has specific tests because it extends an interface that's higher in the hierarchy, which does not make use of `cats.effect.Resource`

File: README.md [hunk 5]
```diff
@@ -131,14 +134,18 @@ object MySuite extends IOSuite {
 
 #### Other suites
 
-Weaver also includes support for `ZIO`-based suites via the optional `weaver-zio` dependency.
+Weaver also includes support for
```
Comment (Baccata) [inline README.md:137]: Thanks for the added bits in the readme. Could you also add a page in the website ? You'll have to : 

* add a new markdown file like [this one]((https://github.com/bfdes/weaver-test/blob/monix-support/docs/zio_usage.md) in the same directory
* reference it  [there](https://github.com/bfdes/weaver-test/blob/master/website/sidebars.json#L15 )

You can build the website locally following these [instructions](https://scalameta.org/mdoc/docs/docusaurus.html#create-static-site) and call `yarn run start` to automatically open it in your browser

File: modules/monix/src/weaver/monixcompat/suites.scala [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  val scheduler                       = Scheduler(ec)
+  implicit def timer: Timer[Task]     = Task.timer(scheduler)
+  implicit def cs: ContextShift[Task] = Task.contextShift(scheduler)
+  implicit def effect: ConcurrentEffect[Task] =
+    new CatsConcurrentEffectForTask()(scheduler, Task.defaultOptions)
+}
+
+trait MutableMonixSuite extends MutableFSuite[Task] with BaseMonixSuite
```
Comment (Baccata) [inline modules/monix/src/weaver/monixcompat/suites.scala:19]: I'd call these `TaskSuite` for coherence, as the name of the effect type is `Task` (ie the cats-effect specific one is called `IOSuite` as opposed to `CatsSuite`)

**Comment by Baccata** [comment]
Hi @bfdes , thanks a lot for this 😄 . I think it's closer to ready than you might think, just need a page added to the website, and a couple modifications in names and accessors

File: modules/monix/src/weaver/monixcompat/suites.scala (hunk 6 above)
Comment (bfdes) [reply modules/monix/src/weaver/monixcompat/suites.scala:19]: No problem. I was a bit confused as the ZIO suite is called `MutableZIOSuite`, and I thought the ZIO effect type was also called `Task`?

File: modules/monix/src/weaver/monixcompat/suites.scala (hunk 4 above)
Comment (bfdes) [reply modules/monix/src/weaver/monixcompat/suites.scala:11]: 👍 I'll replace it with `import monix.execution.Scheduler.Implicits.global`, which I imagine wraps `scala.concurrent.ExecutionContext.global`.

Comment (Baccata) [reply modules/monix/src/weaver/monixcompat/suites.scala:11]: I'd just do `val scheduler : Scheduler = monix.execution.Scheduler.global`, no need to refer to it implicitly I think, considering it's not used in combinator methods on Task

File: modules/monix/src/weaver/monixcompat/suites.scala (hunk 6 above)
Comment (Baccata) [reply modules/monix/src/weaver/monixcompat/suites.scala:19]: You're not wrong, but `Task[R, A]` is type alias for `ZIO[R, Throwable, A]`, so the actual effect type is called `ZIO`

File: modules/monix/test/src/weaver/monixcompat/MonixSuiteTest.scala (hunk 3 above)
Comment (bfdes) [reply modules/monix/test/src/weaver/monixcompat/MonixSuiteTest.scala:12]: I found it a bit hard to follow the base framework tests, so I wasn't sure. 
In that case, I can replace `MonixSuite` here with `SimpleMonixSuite` here. Since that is essentially what it is.

File: modules/monix/src/weaver/monixcompat/suites.scala (hunk 4 above)
Comment (bfdes) [reply modules/monix/src/weaver/monixcompat/suites.scala]: Addressed in b644c8a823f102f8bcc638299adb09818008ed9c.

File: modules/monix/src/weaver/monixcompat/suites.scala (hunk 6 above)
Comment (bfdes) [reply modules/monix/src/weaver/monixcompat/suites.scala]: Addressed in b644c8a823f102f8bcc638299adb09818008ed9c.

File: README.md (hunk 5 above)
Comment (bfdes) [reply README.md:137]: Addressed in 4b29674bf0724234761a5a08f6337557ec19de7f.

**Review by Baccata — APPROVED** [review]

**Comment by Baccata** [comment]
Thanks a lot for this @bfdes !
